tools/gn

```diff
@@ -208,52 +208,59 @@ def get_target_cpu(args):
   return 'x64'
 
 
-def to_gn_args(args):
-  if args.simulator:
-    if args.target_os != 'ios':
-      raise Exception('--simulator is only supported for iOS')
-
-  runtime_mode = args.runtime_mode
-
-  gn_args = {}
-
-  gn_args['is_debug'] = args.unoptimized
+def setup_goma(args):
+  goma_gn_args = {}
 
   goma_dir = os.environ.get('GOMA_DIR')
   goma_home_dir = os.path.join(os.getenv('HOME', ''), 'goma')
-  depot_tools_bin_dir = os.path.join(args.depot_tools, '.cipd_bin')
 
   # GOMA has a different default (home) path on gWindows.
   if not os.path.exists(goma_home_dir) and sys.platform.startswith(
       ('cygwin', 'win')):
     goma_home_dir = os.path.join('c:\\', 'src', 'goma', 'goma-win64')
 
   if args.target_os == 'wasm' or args.web:
-    gn_args['use_goma'] = False
-    gn_args['goma_dir'] = None
+    goma_gn_args['use_goma'] = False
+    goma_gn_args['goma_dir'] = None
     print('Disabling GOMA for wasm builds, it is not supported yet.')
-  elif args.goma and goma_dir:
-    gn_args['use_goma'] = True
-    gn_args['goma_dir'] = goma_dir
+  elif args.goma and goma_dir and os.path.exists(goma_dir):
+    goma_gn_args['use_goma'] = True
+    goma_gn_args['goma_dir'] = goma_dir
   elif args.goma and os.path.exists(goma_home_dir):
-    gn_args['use_goma'] = True
-    gn_args['goma_dir'] = goma_home_dir
-  elif args.goma and os.path.exists(depot_tools_bin_dir):
-    gn_args['use_goma'] = True
-    gn_args['goma_dir'] = depot_tools_bin_dir
+    goma_gn_args['use_goma'] = True
+    goma_gn_args['goma_dir'] = goma_home_dir
+  elif args.goma:
+    raise Exception(
```

Comment:
Previously, there was a fallback to non-goma builds if GOMA could not be found, rather than a hard failure.
This change to a hard failure may break some CI systems.